parallel_gripper_controller/include/parallel_gripper_controller/parallel_gripper_action_controller_impl.hpp

```diff
@@ -216,7 +216,7 @@ void GripperActionController::check_for_success(
   double current_velocity)
 {
   RealtimeGoalHandlePtr active_goal;
-  rt_active_goal_.get([&](const RealtimeGoalHandlePtr & goal) { active_goal = goal; });
+  rt_active_goal_.try_get([&](const RealtimeGoalHandlePtr & goal) { active_goal = goal; });
   if (!active_goal)
```

Comment:
```suggestion
  const bool got_goal =
    rt_active_goal_.try_get([&](const RealtimeGoalHandlePtr & goal) { active_goal = goal; });
  if (!got_goal || !active_goal)
```
something like this. but if the try_set fails, we can't do much about it. it will just be re-evaluated at the next update call.

Reply:
Would you mind if we talked this through a bit please?

When `rt_active_goal_.try_get([&](const RealtimeGoalHandlePtr & goal) { active_goal = goal; }); `returns false, it means the operation failed to acquire the lock on the realtime-safe container without blocking.

In this specific context (the `check_for_success` function), if `try_get` returns false, the `active_goal` variable remains uninitialized (or rather, it remains as a default-constructed RealtimeGoalHandlePtr, which would be a null pointer).

The code correctly handles this scenario with the following check:
```
if (!active_goal)
{
  return;
}
```

Now...

I'm more worried about the if the `try_set` fails, the goal remains active in `rt_active_goal_`, and on the next control cycle, `check_for_success()` will be called again with the same goal that was already marked as succeeded/aborted. This could lead to duplicate completion notifications - calling setSucceeded/setAborted multiple times on the same goal.

Then I was thinking maybe only call `setSucceeded/setAborted` only when the try_set succeeds:
```
  if (fabs(error_position) < params_.goal_tolerance)
  {
    pre_alloc_result_->state.effort[0] = computed_command_;
    pre_alloc_result_->state.position[0] = current_position;
    pre_alloc_result_->reached_goal = true;
    pre_alloc_result_->stalled = false;
    
    // Only complete the goal if we can also clear it atomically
    if (rt_active_goal_.try_set([](RealtimeGoalHandlePtr & stored_value)
                                { stored_value = RealtimeGoalHandlePtr(); }))
    {
      RCLCPP_DEBUG(get_node()->get_logger(), "Successfully moved to goal.");
      active_goal->setSucceeded(pre_alloc_result_);
    }
    else
    {
      RCLCPP_DEBUG(get_node()->get_logger(), "Goal reached but cleanup failed, will retry next cycle");
    }
  }
```

But the problem of course here is that we are delaying goal completion notification until the `try_set()` succeeds.

Reply:
> In which cases try_set will fail? When any other blocking set/get command is imminent. This means that if we cancelled/aborted the current goal in check_for_success, then it is very likely that the content of rt_active_goal_ will be different in the next update call? This would mean that a failing try_set is fine here?

Oh yes! You have a point, I din't think it through. Yes, failing `try_set` is fine then. I don't think we need to do anything more. What do you think?

If all is well, all the comments can be resolved?